src/msg/consumer/options.go

```diff
@@ -164,6 +166,16 @@ func (opts *serverOptions) SetConsumeFn(value ConsumeFn) ServerOptions {
 	return &o
 }
 
+func (opts *serverOptions) MessageFn() MessageFn {
+	return opts.messageFn
+}
+
+func (opts *serverOptions) SetMessageFn(value MessageFn) ServerOptions {
+	o := *opts
+	o.messageFn = value
```

Comment:
What's the distinction between `consumeFn` and `messageFn`? What's `messageFn` intended to do?

Reply:
Both `consumeFn` and `messageFn` needs to be implemented by user, if the user wants to process the consumer then they can pass in a `consumeFn`, this is useful when users can reuse resource on consumer level. When users don't care about controlling the consumer, they could simply pass in a `messageFn` that processes the message.

Reply:
Ah ok cool, thanks for the explanation! Maybe add to the docs to explain when to use `consumerFn` vs `messageFn`? I think that would get at what I was asking.

Reply:
Sounds good, added more comments